### DISCUSSION TITLE
Component missing from front end layout manager

### DISCUSSION THREAD

**Discussion by Tigerlady** [MasaCMS/MasaCMS#76] [General]
I either missed this or it got overlooked but the menu item for components is missing in the front-end layout manager.  
OLD MURA Site
![image](https://user-images.githubusercontent.com/1850313/164511220-fd5c07c1-9fc7-40ed-bb69-aaf87d157254.png)

MASA SITE
![image](https://user-images.githubusercontent.com/1850313/164511382-93ee87a6-91da-47cc-b666-3fb1585ac9f4.png)

Is there a way to get this back?  I am converting a Mura site to Masa and I use components in a lot of places that I do not want to have to rewrite if possible.

**Comment by guustnieuwenhuis** [comment]
We're working a various improvements to the modules. But for the mean time, you can drag a component to the page using the text module and select a component as content source.

**Comment by Tigerlady** [comment]
Guust,

Thank you so much.  That helps a lot.  


Barbara Renowden
Centric Web, Inc.
19W169 Millbrook Court
Downers Grove, Il 60516
630-541-9956
***@***.***

**Comment by ptutty-wbs** [comment]
I've hidden certain module menu items from users by moving them out of /core/modules/v1/

is there a better way of doing this i.e. a config file somewhere?

**Reply by no3rdw** [reply]
I realize I'm replying months after your post, but in case anyone still finds this helpful, to hide a module from the front end editor without modifying the core, you need to extend the config.xml.cfm file into your site or theme directory and then change the 'contenttypes' parameter from '*' to ''.

For example, to hide the 'collection' module, you would create a folder named 'collection' under the 'modules' folder either in your site or theme directory, then create a file called 'config.xml.cfm' in that collection folder, with the contents: 
`<mura name="Collection" contenttypes="" iconclass="mi-th-list"/>
`

More details in the Mura docs, here: https://docs.murasoftware.com/v10/extending-mura/mura-rendering/#section-elements-of-the-configxmlcfm-file

**Comment by kerryhill** [comment]
Has "Component" been added to the Modules yet?

**Reply by no3rdw** [reply]
As Guust said above, the 'Component' functionality that Mura displayed as a module is still present under the 'Text' module - you just have to set the module's content source to 'Component' and choose your component.  Is there some older functionality that you're looking for that is missing from using this method?

**Reply by kerryhill** [reply]
Well, since non-coders will use this functionality it should be as straightforward and simple as possible. I don't want to have to explain a workaround.  Components being an important part of the application I'm just surprised it's not there. Do you know any way I could add it?

**Reply by guustnieuwenhuis** [reply]
Hi @kerryhill,
The reason why we opt not to add a separate module only to select a component is to keep things simple. A component is a pre-defined, reusable block of content. We believe the option to add it as a content source through a text module is the most best way.

But you don't have to agree with us on this. :)

You can create your own version of a module that has the name and functionality you desire. Documentation for that can be found in the Mura docs:
- https://docs.murasoftware.com/v7-1/mura-developers/mura-rendering/#section-mura-modules-display-objects
- https://docs.murasoftware.com/v10/extending-mura/#section-mura-modules
